pkg/http/sender.go

```diff
@@ -0,0 +1,605 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package http
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/questdb/go-questdb-client/v3/pkg/buffer"
+	"github.com/questdb/go-questdb-client/v3/pkg/conf"
+)
+
+type tlsMode int64
+
+const (
+	tlsDisabled           tlsMode = 0
+	tlsEnabled            tlsMode = 1
+	tlsInsecureSkipVerify tlsMode = 2
+)
+
+var (
+	// We use a shared http transport to pool connections
+	// across LineSenders
+	globalTransport http.Transport = http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		MaxConnsPerHost:       0,
+		MaxIdleConns:          100,
+		MaxIdleConnsPerHost:   100,
+		IdleConnTimeout:       90 * time.Second,
```

Comment:
Same here and for the below settings: we need to let users configure them.

Reply:
I had problems writing integration tests with the global transport since there was a dependency on test ordering (to ensure that `TlsMode` was not used after `TlsInsecureVerify` was).  So I added a transport override option in [this commit](https://github.com/questdb/go-questdb-client/pull/26/commits/7b2263b4ba581ce55a71bc694c0f1fed4534123f). WDYT?

Reply:
Looks like a flexible enough option for folks who want to tweak the transport. 👍